adobe/react-spectrum
# fix(#2766): Virtualizer does not scroll items into view within the browser viewport

Add IntersectionObserver to detect whether or not a focused item is visible within the browser window, and if not scroll it into view.


Closes #2766 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue #2766.

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/d5eb632b807c8dda8faea50e3cbcc5995528d6bf/storybook/index.html?path=/story/listbox--async-loading&id=listbox--async-loading&providerSwitcher-toastPosition=bottom&viewMode=story
2. Reduce the inner height of the browser window to be less than the height of the ListBox within the story, ~200px should be good.
3. <kbd>Tab</kbd> to set keyboard focus to the first Item within the ListBox.
4. Use <kbd>ArrowDown</kbd> to scroll until you focus an Item within the ListBox that is outside of the browser viewport.
5. Notice that the browser will now automatically scroll the `document.body` to bring the focused Item into view.

Other stories to test: 

- https://reactspectrum.blob.core.windows.net/reactspectrum/d5eb632b807c8dda8faea50e3cbcc5995528d6bf/storybook/index.html?path=/story/tableview--async-loading&id=tableview--async-loading&providerSwitcher-toastPosition=bottom&viewMode=story
- https://reactspectrum.blob.core.windows.net/reactspectrum/d5eb632b807c8dda8faea50e3cbcc5995528d6bf/storybook/index.html?path=/story/useselectablelist--static-ul-static-sub-ul&id=useselectablelist--static-ul-static-sub-ul&providerSwitcher-toastPosition=bottom&viewMode=story
- https://reactspectrum.blob.core.windows.net/reactspectrum/d5eb632b807c8dda8faea50e3cbcc5995528d6bf/storybook/index.html?path=/story/focusscope--focus-safely&id=focusscope--focus-safely&providerSwitcher-toastPosition=bottom&viewMode=story

## 🧢 Your Project:
Adobe/Accessibility Adobe/Admin_Console Jira issues A11Y-5009 and TRON-8092
